src/encoded/static/components/award.js

```diff
@@ -409,24 +737,481 @@ class StatusChart extends React.Component {
+        const anisogenicFacet = anisogenic.facets.find(facet => facet.field === 'status');
+        unreplicatedArray = (unreplicatedFacet && unreplicatedFacet.terms && unreplicatedFacet.terms.length) ? unreplicatedFacet.terms : [];
+        isogenicArray = (isogenicFacet && isogenicFacet.terms && isogenicFacet.terms.length) ? isogenicFacet.terms : [];
+        anisogenicArray = (anisogenicFacet && anisogenicFacet.terms && anisogenicFacet.terms.length) ? anisogenicFacet.terms : [];
+    }
+    const labels = ['proposed', 'started', 'submitted', 'released', 'deleted', 'replaced', 'archived', 'revoked'];
+    if (unreplicatedArray.length) {
+        for (let j = 0; j < labels.length; j += 1) {
+            for (let i = 0; i < unreplicatedArray.length; i += 1) {
+                if (unreplicatedArray[i].key === labels[j]) {
+                    unreplicatedLabel.push(unreplicatedArray[i].key);
+                    unreplicatedDataset.push(unreplicatedArray[i].doc_count);
+                }
+            }
+        }
+        for (let j = 0; j < labels.length; j += 1) {
+            if (labels[j] !== unreplicatedLabel[j]) {
+                unreplicatedLabel.splice(j, 0, labels[j]);
+                unreplicatedDataset.splice(j, 0, 0);
+            }
+        }
+    } else {
+        unreplicatedDataset = [0, 0, 0, 0, 0, 0, 0, 0];
+    }
+    if (isogenicArray.length) {
+        for (let j = 0; j < labels.length; j += 1) {
+            for (let i = 0; i < isogenicArray.length; i += 1) {
+                if (isogenicArray[i].key === labels[j]) {
+                    isogenicLabel.push(isogenicArray[i].key);
+                    isogenicDataset.push(isogenicArray[i].doc_count);
+                }
+            }
+        }
+        for (let j = 0; j < labels.length; j += 1) {
+            if (labels[j] !== isogenicLabel[j]) {
+                isogenicLabel.splice(j, 0, labels[j]);
+                isogenicDataset.splice(j, 0, 0);
+            }
+        }
+    } else {
+        isogenicDataset = [0, 0, 0, 0, 0, 0, 0, 0];
+    }
+    if (anisogenicArray.length) {
+        for (let j = 0; j < labels.length; j += 1) {
+            for (let i = 0; i < anisogenicArray.length; i += 1) {
+                if (anisogenicArray[i].key === labels[j]) {
+                    anisogenicLabel.push(anisogenicArray[i].key);
+                    anisogenicDataset.push(anisogenicArray[i].doc_count);
+                }
+            }
+        }
+        for (let j = 0; j < labels.length; j += 1) {
+            if (labels[j] !== anisogenicLabel[j]) {
+                anisogenicLabel.splice(j, 0, labels[j]);
+                anisogenicDataset.splice(j, 0, 0);
+            }
+        }
+    } else {
+        anisogenicDataset = [0, 0, 0, 0, 0, 0, 0, 0];
+    }
+    return ([labels, unreplicatedLabel, unreplicatedDataset, isogenicDataset, anisogenicDataset]);
+}
+
+class ControlsChart extends React.Component {
```

Comment:
This returns an array of arrays. To avoid the use of “magic” numbers in places that call StatusData, I’d like to see returning an object so that the different data can be extracted by key instead of with magic numbers.